plugins/orchestrator-backend/src/service/DataIndexService.ts

```diff
@@ -147,6 +154,24 @@ export class DataIndexService {
     return processInstances;
   }
 
+  public async getProcessInstancesTotalCount(): Promise<number> {
+    const graphQlQuery = buildGraphQlQuery({
+      type: 'ProcessInstances',
+      queryBody: 'id',
+    });
+    this.logger.debug(`GraphQL query: ${graphQlQuery}`);
+    const result = await this.client.query(graphQlQuery, {});
+
+    if (result.error) {
+      this.logger.error(`Error when fetching instances: ${result.error}`);
+      throw result.error;
+    }
+
+    const idArr = result.data.ProcessInstances as ProcessInstance[];
```

Comment:
If you're already getting all of the instances why not just do the pagination logic here? 
I'm concerned this is making things slower... the whole idea of pagination is to make things faster

Reply:
@caponetto 
WDYT

Reply:
can you please check what the difference in payload is?

Reply:
It depends on the number of **all instances** and the **process definition structure**. When the number of nodes is high, the instance will have a larger JSON output. When the result payload is larger, it can lead to [graphQL query complexity](https://hygraph.com/docs/api-reference/basics/query-complexity). It will result in a runtime error. 

However, if we have the `totalCount` as a result output[1] in the graphQL, we can remove this method. 

[1] https://github.com/apache/incubator-kie-kogito-apps/issues/2012

Here is a simple comparison which I did with the default orchestrator container we use in our dev setup. I have mostly executed the Hello world workflow. I have 12 workflow instances.

- GraphQL query without pagination params(returns all instance data) - 2.76KB
- GraphQL query only to fetch all IDs - 450B